subprojects/base-services/src/main/java/org/gradle/api/GradleSystem.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api;
+
+import org.gradle.api.internal.GradleProcessEnvironment;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * Gradle environmental variable collection.
+ *
+ * @since 4.10
+ */
+@Incubating
+public class GradleSystem {
```

Comment:
Did you consider an instance method like `StartParameter#getEnv`? What made you choose a static method instead?

Reply:
I think the transition `System.getenv` -> `GradleSystem.getenv` would make it better for users to understand and use.

Reply:
👍

Reply:
On the other hand it makes the API rather unsafe, and prevents us from doing smarter things in the future like providing different environments to workers. I'd make it an instance API.

Reply:
It doesn't prevent us from providing different environments to workers. The worker API already has a way to mutate the environment on a per-worker basis.